app/src/main/java/com/joocoding/android/app/githubsearch/MainActivity.kt

```diff
@@ -0,0 +1,90 @@
+package com.joocoding.android.app.githubsearch
+
+import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.SearchView
+import androidx.databinding.DataBindingUtil
+import com.joocoding.android.app.githubsearch.databinding.ActivityMainBinding
+import com.joocoding.android.app.githubsearch.model.response.RepositoriesResponse
+import com.joocoding.android.app.githubsearch.model.response.RepositoryResponse
+import com.joocoding.android.app.githubsearch.network.GithubRetrofit
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var mainAdapter: MainAdapter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
+        initRecycler()
+        getRepository(GithubRetrofit.githubRetrofit.searchRepositories())
+        initSearchView()
+        supportActionBar?.hide()
+    }
+
+    private fun initRecycler() {
+        mainAdapter = MainAdapter()
+        binding.recyclerView.adapter = mainAdapter
+
+    }
+
+    private fun getRepository(repositoryRequest: Call<RepositoriesResponse>) {
+        repositoryRequest.enqueue(object : Callback<RepositoriesResponse> {
+            override fun onFailure(call: Call<RepositoriesResponse>, t: Throwable) {
+                Log.e(TAG, "Failed to fetch photos", t)
+            }
+
+            override fun onResponse(
+                call: Call<RepositoriesResponse>,
+                response: Response<RepositoriesResponse>
+            ) {
+                Log.d(TAG, "Response received")
+                val repositoriesResponse: RepositoriesResponse? = response.body()
+                var repositoryResponse: List<RepositoryResponse> =
+                    repositoriesResponse?.repositories
+                        ?: mutableListOf()
+
+                repositoryResponse = repositoryResponse.filterNot {
+                    it.owner.avatarUrl.isBlank()
+                }
```

Comment:
창익님이 조언해주신 대로 변경하면 좋을 것 같아요!
추가로 null일 때, 빈 리스트를 넘겨주기 위해 mutableListOf()를 선언하기 보다는 `emptyList`를 넘겨주는 것이 명시적일 것 같아요!

```suggestion
                val repositoryResponse: List<RepositoryResponse> =
                    repositoriesResponse?.repositories?.filter { owner.avatarUrl.isNotBlank() }
                        ?: empryList()
```